MekHQ/src/mekhq/gui/adapter/FinanceTableMouseAdapter.java

```diff
@@ -72,38 +74,35 @@ public void actionPerformed(ActionEvent action) {
     }
 
     @Override
-    public void mousePressed(MouseEvent e) {
-        maybeShowPopup(e);
+    protected boolean shouldShowPopup() {
+        return financeTable.getSelectedRowCount() > 0;
     }
 
     @Override
-    public void mouseReleased(MouseEvent e) {
-        maybeShowPopup(e);
-    }
+    @Nullable
+    protected JPopupMenu createPopupMenu() {
+        int row = financeTable.getSelectedRow();
+        if (row < 0) {
+            return null;
+        }
```

Comment:
I'm pretty sure this is not necessary with the check in shouldShowPopup.

Reply:
I don't like eschewing preconditions if they're not both private; but I think I can simplify this to a single method.